src/internal.rs

```diff
@@ -462,18 +464,28 @@ impl Pal {
         self.data.is_empty()
     }
 
-    pub fn as_slice<BD: BitDepth>(&self) -> &[[[BD::Pixel; 8]; 3]] {
-        as_chunks::<3, [BD::Pixel; 8]>(
-            as_chunks::<8, BD::Pixel>(BD::cast_pixel_slice(&self.data)).0,
-        )
-        .0
+    pub fn index<'a: 'b, 'b, BD: BitDepth>(
```

Comment:
Don't `index` and `index_mut` still need to be `unsafe`? Safety when indexing into a `DisjointMut` relies on the caller ensuring that the index is actually disjoint from other accesses, so I think helper methods like this that are doing `DisjointMut` access under the hood still need to be `unsafe`

Reply:
If `index_mut` is used properly (i.e. never overlapping any other concurrent slice reference), then we don't. My argument to myself there was that we only need to put the safety comment and unsafe requirements on that method because of that. If one only used `index` there's no way to make that unsafe, you can only make something break here by using `index_mut` incorrectly.

I'm open to making it unsafe, I just don't think it is required.

Reply:
Okay, that makes sense to me wrt `index` not needing to be `unsafe`. But I think we need to keep `index_mut` unsafe. Any case where the function depends on "if it's used properly" is by definition unsafe, I think.

Reply:
Absolutely, I missed that I didn't here, fixing that.